distribution/push.go

```diff
@@ -127,6 +133,19 @@ func Push(ctx context.Context, ref reference.Named, imagePushConfig *ImagePushCo
 			continue
 		}
 
+		parsedURL, urlErr := url.Parse(endpoint.URL)
```

Comment:
I don't really think that adds clarity. The current code parses a url and checks the scheme. This is very clear. No need to be DRY for DRY's sake.